### DISCUSSION TITLE
TUGboat 43:1 article: “Markdown 2.15.0: What's new?”

### DISCUSSION THREAD

**Discussion by Witiko** [Witiko/markdown#122] [General]
By March 31, I would like to submit a [TUGboat][1] article, which will discuss the new features since [the Markdown 2.10.0 article][4]. I already [drafted an outline of the article][2]. Co-authorship by those who contributed the new features is welcome (@drehak, @xvrabcov, @TeXhackse, @michal-h21, @writersglen). Alternatively, if you'd like to contribute to the text in small ways but don't want to commit to full authorship, informal review of the text will also be appreciated. 🙂

 [1]:https://tug.org/tugboat/
 [2]: https://www.overleaf.com/7895438692wphwvndgkdky
 [3]: https://www.overleaf.com/read/ctqnbhtpzrzj
 [4]: https://www.overleaf.com/read/fkbykzqxfjjc

**Comment by writersglen** [comment]
Hello,

I heartily applaud and support as best I can Vit’s initiative to extend the utility of markdown to the TeX community. As I see it, there are three stakeholders in this project: coders engaged in the technical challenges, the TeX community at large, and the many thousands of self-publishers who stand to publish trade-quality books with less sweat and tears.

I fall into the third category. I’ve published three books composed and typeset with LaTeX and am close to launch of a fourth. My insufficient LaTeX fluency has made all four projects more difficult than necessary. For this reason I’m particularly interested in Vit’s book publishing interface proposal. Given my limited TeX competencies, however, the best I can offer is a Mickey-the-Dunce user perspective.

As an end user, the first thing I’d like to see is a clear statement of how the book publishing interface can help me publish better books. The second thing is clear and comprehensive how-to documentation. In short, I need confidence that adoption of TeX markdown shortcomings will not leave me in a lurch if I choose to adopt it into my workflow.

Much work needs to be done, seems to me, to accomplish this goal. First is a statement of intent that inspires all stakeholders to put their shoulders behind the project. Second is a definitive list of markdown tags necessary and sufficient to meet the diverse needs of book publishers. This list will require input and consensus of all stakeholders—but may not be an easy thing to achieve.

I’d be happy to add my evolving wish list to the functional necessities of the book publishing interface as well as help with end-user documentation. I can also offer modest bounties for accomplishment of well-specified technical milestones.

Such is my belief that book publishing-focused markdown is a significant and consequential contribution to the TeX ecosystem.

Lloyd R. Prentice
Writersglen Publications

**Comment by Witiko** [comment]
Dear Karl,

my contributors and I are planning to submit an article to TUGboat 43:1.
Is it OK if we submit by March 31, or would you rather if we submit our
article in advance -- say during the next week?

Best,  
Vitek

**Comment by Witiko** [comment]
my contributors and I are planning to submit an article to TUGboat 43:1.
    Is it OK if we submit by March 31, or would you rather if we submit our
    article in advance -- say during the next week?

It's ok to submit March 31, but earlier is always appreciated, when
possible. I.e., when an article is ready, go ahead and send it in
anytime. Looking forward. Thanks! -k

**Comment by Witiko** [comment]
@drehak @xvrabcov @michal-h21 I have finished the introductory text that describes the three sections of [the article][1]. For each subsection, I added relevant links and information about who is responsible for writing and reviewing the subsection. The expected structure of each subsection is introduction, examples, conclusion (optional). Here is a minimal example of a subsection:

![task-lists](https://user-images.githubusercontent.com/603082/160296935-f79e837f-b9c9-4e41-883a-b62d92f2b837.png)

I would like to submit the article by Tuesday (March 29). By then, the text should be written and reviewed.

 [1]: https://www.overleaf.com/7895438692wphwvndgkdky

**Reply by Witiko** [reply]
Note that this is a subsection for the Writer's newsletter section, which aims at writers and is the least technical section of the article. Examples in the Coder's newsletter and in the Developer's newsletter will likely be less visual and more code-centric.

**Comment by Witiko** [comment]
@drehak @xvrabcov @michal-h21 We are running slightly behind schedule, but I have just [finished Section 1][1] and I expect to have a finished copy ready in the evening. When I have finished, I will let you know and I will then submit the article to Karl Berry at midnight (CEST), your review by then will be appreciated. We will have a chance to confirm the final version before it appears in TUGboat, but it would be best if we make our edits before the submission. We can use the Overleaf chat for quick communication.

 [1]: https://www.overleaf.com/2924942886vffrhgmdbtrh

**Reply by Witiko** [reply]
I just finished proofreading and copy-editing [the article][1]. I will appreciate your review by 12PM (CEST), then I will submit the article to Karl Berry.

 [1]: https://www.overleaf.com/2924942886vffrhgmdbtrh

**Comment by Witiko** [comment]
Dear Karl,

attached, you will find our submission for TUGboat 43:1. For the printed
version, you will want to uncomment the \usemintedstyle{bw} command on
line 59 to disable colorful syntax highlighting in listings.

Best,
Vit

**Reply by Witiko** [reply]
Attachments:
- [Markdown_2_15_0__What_s_new_.pdf](https://github.com/Witiko/markdown/files/8388805/Markdown_2_15_0__What_s_new_.pdf)
- [Markdown_2_15_0__What_s_new_.zip](https://github.com/Witiko/markdown/files/8388806/Markdown_2_15_0__What_s_new_.zip)

**Comment by writersglen** [comment]
Hi Vit,

Nice work!

I'm pleased that markdown 2.15.0 recognizes that needs of tech-wary self-publishers differ to some extent from traditional LaTeX users and developers.

1.3 Cross-references and 1.5 LATEX themes for self-publishers are much welcomed.

It's not clear if 2.2 Passing HTML through to tex4ht, 3.2 Attributes on links and images, and 3.3 Importing LATEX setup snippets are aspirational or actually production ready, but they certainly promise to make LaTeX typography more functional for book publishers.

Anything that can be done to make development of custom-defined renderers easier for non-coders will add significant power to your package. I don't at this time understand enough to benefit. But the Exending Markdown chapter in *Beautiful* could be expanded with more tutorial content to make the feature more available to a wider constituency. But I would need tutoring. This might also be the basis for another TUG article.

One tag that would certainly be useful in my limited experience is an implimentation of LaTeX's vspace command.

While waiting for Tereza's work on *Beautiful,* I've been working on my next book, *Precision Marketing for Self-Publishers.* *Precision* could  provide yet another test bed for the applicability of markdown package to self-publishing community.

All the best,

Lloyd

**Comment by Witiko** [comment]
@xvrabcov @drehak @michal-h21 Karl Berry sent us [proofs of the TUGboat article][1]. If you discover any issues with the text that you'd like to see fixed, please let me know by tomorrow.

 [1]: https://tug.org/TUGboat/production/43-1/tb133novotny-markdown.pdf

**Reply by Witiko** [reply]
For example, I would like to change "In our proof-of-concept" to "In our proof of concept" in the first paragraph of Section 2.3.
